patchwork/common/tools/api_tool.py

```diff
@@ -0,0 +1,124 @@
+import json
+from typing import Any, Dict, Optional
+
+import requests
+from typing_extensions import Literal
+
+from patchwork.common.tools.tool import Tool
+
+
+class APIRequestTool(Tool, tool_name="make_api_request", abc_register=False):
+    __base_url = ""
+    __headers = dict()
+    __auth = None
+    __data_prefix = ""
+
+    def __init__(
+        self,
+        base_url: Optional[str] = None,
+        headers: Optional[Dict[str, str]] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        data_prefix: Optional[str] = None,
+        **kwargs,
+    ):
+        if base_url:
+            self.__base_url = base_url
```

Comment:
base_url is conflicting, the tool description mention that the url should be the full url but the presence of this base_url indicate that sometimes it should be a url path instead.